src/com/serotonin/mango/rt/event/detectors/AlphanumericStateDetectorRT.java

```diff
@@ -38,10 +38,26 @@ public LocalizableMessage getMessage() {
         String prettyText = vo.njbGetDataPoint().getTextRenderer().getText(vo.getAlphanumericState(),
                 TextRenderer.HINT_SPECIFIC);
         LocalizableMessage durationDescription = getDurationDescription();
+        String description = (vo.njbGetDataPoint().getDescription() == null || vo.njbGetDataPoint().getDescription().equals("")) ? "" : " (" + vo.njbGetDataPoint().getDescription() + ")";
+        String eventRendererText = (vo.njbGetDataPoint().getEventTextRenderer() == null) ? "" : vo.njbGetDataPoint().getEventTextRenderer().getText(vo.getAlphanumericState());
```

Comment:
I would suggest moving this description and eventRendererText extraction to a separate method.
It looks like this is code in many places, you can create a util class for this.